internal/ethapi/api.go

```diff
@@ -1520,6 +1520,13 @@ func (args *SendTxArgs) toTransaction() *types.Transaction {
 
 // SubmitTransaction is a helper function that submits tx to txPool and logs a message.
 func SubmitTransaction(ctx context.Context, b Backend, tx *types.Transaction) (common.Hash, error) {
+	// If the transaction fee cap is already specified, ensure the
+	// fee of the given transaction is _reasonable_.
+	feeEth := new(big.Float).Quo(new(big.Float).SetInt(new(big.Int).Mul(tx.GasPrice(), new(big.Int).SetUint64(tx.Gas()))), new(big.Float).SetInt(big.NewInt(params.Ether)))
+	feeFloat, _ := feeEth.Float64()
+	if b.RPCTxFeeCap() != 0 && feeFloat > b.RPCTxFeeCap() {
+		return common.Hash{}, fmt.Errorf("txfee(%.2f ether) exceeds the maximum setting(%.2f ether)", feeFloat, b.RPCTxFeeCap())
```

Comment:
I've noticed that throughout your PRs you started omitting a space character from front of parantheses. Please add that (i.e. `tx fee<space>(%.2f ether) exceeds...`)

Reply:
Also instead of `maximum setting` let's write `configured cap`.

Reply:
Done